### ISSUE TITLE
Unreasonable results on image classification when using pre-trained weights

### ISSUE THREAD

**Issue by ywpkwon** [tensorflow/models#3974]
I am checking the stock form of Tensorflow Slim models. I downloaded the pre-trained weights, e.g., `vgg_16`, `vgg_19`, `inception_v1`, `inception_v2`.

In my following minimal codes, I can choose (and load) a network and perform image classification. When I set `name = 'vgg_16'` or `name = 'vgg_19'`, the results are okay. **However, `inception_v1` and `inception_v2` return entirely wrong results.** (I am using very easy and obvious dog, cat or elephant images).

* The codes have `num_classes` parameter, and `vgg`s should be 1000 and `inception`s should be 1001. (https://github.com/tensorflow/models/tree/master/research/slim#the-resnet-and-vgg-models-have-1000-classes-but-the-imagenet-dataset-has-1001) I did it correctly and, since the result is just 1 index offset, I also tried printing `argmax` index and its neighbors. But they are all entirely wrong. (eg. dog -> apron). 

I feel like it seems not the matter of accuracy. Is there any part that I am missing here?

    import os
    import sys
    import json
    import imageio
    import tensorflow as tf
    import numpy as np

    sys.path.append('../slim')
    from nets import nets_factory
    from preprocessing import preprocessing_factory

    slim = tf.contrib.slim

    # load labels
    with open('imagenet_class_index.json') as f:
        labeltxt = json.load(f)
        idx2label = [labeltxt[str(k)][1] for k in range(len(labeltxt))]

    name = 'inception_v2'
    checkpoints_dir = '../slim/slim-pretrained'

    with tf.Graph().as_default():

        # network and preprocessing
        network_fn = nets_factory.get_network_fn(name, num_classes=1001, is_training=False)
        image_preprocessing_fn = preprocessing_factory.get_preprocessing(name, is_training=False)
        image_size = network_fn.default_image_size

        # image tensor
        image_tensor_ph = tf.placeholder(tf.float32, (None, None, 3))    
        image_tensor = image_preprocessing_fn(image_tensor_ph, image_size, image_size)
        image_tensor = tf.expand_dims(image_tensor, 0)

        logits, _ = network_fn(image_tensor)
        probabilities_tensor = tf.nn.softmax(logits)

        sess = tf.Session()
        checkpoint_file = os.path.join(checkpoints_dir, '%s.ckpt' % name)
        variables_to_restore = slim.get_variables_to_restore()
        saver = tf.train.Saver(variables_to_restore)
        saver.restore(sess, checkpoint_file)

        x = imageio.imread('dog.jpg')
        idx = np.argmax(sess.run(probabilities_tensor, feed_dict={image_tensor_ph: x}))
        print(idx2label[idx-1:idx+2])

* Just in case, I also post the same question in Stackoverflow (https://stackoverflow.com/questions/49825091/unreasonable-result-on-image-classification-when-using-tensorflow-slims-pre-tra).

**Comment by ywpkwon** [comment]
Oh, nevermind. I found the reason.
`image_tensor_ph = tf.placeholder(tf.float32, (None, None, 3))` should be 
`image_tensor_ph = tf.placeholder(tf.uint8, (None, None, 3))`. 

If I set the placeholder as `float32`, the inception preprocessing functions does not work as expected. (`tf.image.convert_image_dtype`)

**Comment by ywpkwon** [comment]
@yicheng-li  Yes, but in this simple test, I just tested a few easy images and checked if results seemed acceptable.
